py2hwsw/scripts/iob_port.py

```diff
@@ -142,4 +142,5 @@ def create_port(core, *args, signals=[], **kwargs):
         error_msg=f"Invalid {kwargs.get('name', '')} port attribute '[arg]'!",
     )
     port = iob_port(*args, signals=sig_obj_list, interface=interface_obj, **kwargs)
+    replace_duplicate_signals_by_references(core.wires + core.ports, port.signals)
```

Comment:
```suggestion
    replace_duplicate_signals_by_references(core.ports, port.signals)
```
I may be missing something, but I think a port will never want to reference a signal from a wire.

I can see two cases:
1) The wire being referenced is itself a reference to another port.
    In this case, we can just reference to the original port. No need to search for it in wires list.
2) The wire being referenced defines the original signal.
    In this case, the signal of the port will just be a reference to the wire, and this means that the port will not be generated since this port will not be of type `iob_signal`. Check the line 43 of the `io_gen.py` code [here](https://github.com/IObundle/py2hwsw/pull/142/files#diff-79feb0ce0711afc10b8b3250ae7944f2cb8169b1129ad7fdba085cecb37b9206R43).
  This means that, in the generated Verilog, we will just have a wire without the port for it being generated.

Reply:
As an alternative, we could try to fix 2), to allow the port to be generated even if the wire has the original signal.
Maybe by modifying the `replace_duplicate_signals_by_references` to ensure that the original signal is always in the port and not in the wire. This way, this function would move the original signal from the wire to the port when needed, and place the reference just in the wire.

Reply:
I may have haded the wire list by accident, when copying from iob_wire.py.
I don't think there is an advantage to turning existing wires into ports.
However, if one appears, the sugested alternative seems viable.